Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -39,16 +39,71 @@ void EncodeObject(pb_ostream_t* stream,
 std::map<std::string, FieldValue> DecodeObject(pb_istream_t* stream);
 
 /**
- * Note that (despite the value parameter type) this works for bool, enum,
- * int32, int64, uint32 and uint64 proto field types.
- *
- * Note: This is not expected to be called direclty, but rather only via the
- * other Encode* methods (i.e. EncodeBool, EncodeLong, etc)
- *
- * @param value The value to encode, represented as a uint64_t.
+ * Docs TODO(rsgowman). But currently, this just wraps the underlying nanopb
+ * pb_ostream_t. Eventually, this might use static factory methods to create the
+ * underlying pb_ostream_t rather than directly passing it in.
  */
-void EncodeVarint(pb_ostream_t* stream, uint64_t value) {
-  bool status = pb_encode_varint(stream, value);
+class PbOstream {
```

Comment:
Nit: this name just mirrors the Nanopb C-ish convention. I don't have a good suggestion ready, though, so feel free to leave as-is. In particular, I think `Pb` part is unnecessary, and it's a little too abbreviated. `OutputStream`, maybe?

Reply:
In my view, the protobuf part is kinda implied because it's in `serializer`, but I don't feel strongly.

Reply:
`WriteStream`? `Writer`?

This type is entirely local to the serializer so I don't think there's a whole lot of benefit to being super verbose here.

Reply:
FWIW, I like `Writer`.

Reply:
Suggestions are:
- PbOstream, OutputStream, WriteStream, Writer

I'll keep the 'Stream' part, since nobody seems to hate that.

Output vs Writer: I don't care. I don't obviously prefer Writer, so I guess I'll stick with Output. (If you prefer Writer to Output, i.e. you weren't just floating another option, then I'll use Writer.)

Final name (for now): OutputStream.

Reply:
Update: Costa likes Writer. That's two votes for Writer, so I'll use that.